tests/containers/podman_quadlet.pm

```diff
@@ -48,17 +63,38 @@ TimeoutStartSec=120
 WantedBy=multi-user.target
 _EOF_
 
-    my $systemd_vol = <<_EOF_;
+my @systemd_volume = ("$quadlet_dir/$unit_name.volume", <<_EOF_);
 [Volume]
 User=root
 Group=root
 Label=org.test.Key=TESTING
 _EOF_
 
+my @files = (\@systemd_build, \@containerfile, \@systemd_network, \@systemd_pod, \@systemd_container, \@systemd_volume);
+my @units = map { "$unit_name$_.service" } ("", "-pod", "-build", "-network", "-volume");
+
+sub check_unit_states {
+    my $expected = shift // 'generated';
+    foreach my $unit (@units) {
+        validate_script_output("systemctl --no-pager is-enabled $unit", qr/$expected/, proceed_on_failure => 1);
+    }
+}
+
+sub run {
+    my ($self, $args) = @_;
+
+    select_serial_terminal;
+    my $podman = $self->containers_factory('podman');
+    $podman->cleanup_system_host();
+
+    my $quadlet = '/usr/libexec/podman/quadlet';
+
     # create files for generator
     assert_script_run("$quadlet -version");
-    assert_script_run("echo '$_' >> /etc/containers/systemd/$unit_name.container") foreach (split /\n/, $systemd_unit);
-    assert_script_run("echo '$_' >> /etc/containers/systemd/$unit_name.volume") foreach (split /\n/, $systemd_vol);
+    for my $file (@files) {
+        my ($path, $content) = @$file;
+        assert_script_run("echo '$_' >> $path") foreach (split /\n/, $content);
```

Comment:
This would not overwrite an existing file.

Either we add a `script_run("rm -f '$path'");` before this or perhaps try to use `echo -e` and see if that makes the `split /\n/` obsolete.